### PR TITLE
Fleet Placer pull request

### PR DESCRIPTION
**Pull request by xKoem**
Created FleetPlacer microservice with tests.

### CODE REVIEW

**Comment by boczkas** [comment]
Build 138 Still Failing

File: FleetPlacer/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/event/FleetPlacedEvent.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import org.springframework.context.ApplicationEvent;
+import pl.jakubowskiprzemyslaw.tajgertim.models.board.FleetPlacement;
+
+import java.util.Objects;
+
+public class FleetPlacedEvent extends ApplicationEvent {
+
+    private static final long serialVersionUID = -6094656042364570599L;
+    private FleetPlacement fleetPlacement;
```
Comment (Hoymm) [inline FleetPlacer/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/event/FleetPlacedEvent.java]: final

File: FleetPlacer/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/event/FleetPlacedEvent.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package pl.jakubowskiprzemyslaw.tajgertim.fleetplacer.event;
+
+import org.springframework.context.ApplicationEvent;
+import pl.jakubowskiprzemyslaw.tajgertim.models.board.FleetPlacement;
+
+import java.util.Objects;
+
+public class FleetPlacedEvent extends ApplicationEvent {
```
Comment (Hoymm) [inline FleetPlacer/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/event/FleetPlacedEvent.java]: parameterized class for all events

File: FleetPlacer/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/event/FleetPlacedEventListener.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    @Override
+    public void onApplicationEvent(FleetPlacedEvent fleetPlacedEvent) {
+        FleetPlacement fleetPlacement = fleetPlacedEvent.getFleetPlacement();
+
+        queueService.sendObjectToQueue(Queues._7BoardHandlerFleetPlacementQueue, fleetPlacement);
+
+        amountOfFleetsConfigured++;
+        if (amountOfFleetsConfigured == 2) {
```
Comment (Hoymm) [inline FleetPlacer/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/event/FleetPlacedEventListener.java]: use "amountOfFleetsConfigured % 2 == 0" and remove "amountOfFleetsConfigured = 0;"

Comment (Hoymm) [inline FleetPlacer/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/event/FleetPlacedEventListener.java]: extract if to new method

File: FleetPlacer/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/listener/FleetPlacerQueueListener.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Autowired
+    public FleetPlacerQueueListener(LoggerService logger, ApplicationEventPublisher publisher) {
+        this.logger = logger;
+        this.publisher = publisher;
+    }
+
+    @RabbitListener(queues = "FleetPlacementQueue")
+    public void listenOnFleetPlacementQueue(FleetPlacement fleetPlacement) {
+        logger.logInfo(FleetPlacerQueueListener.class, "Received message" + fleetPlacement);
```
Comment (Hoymm) [inline FleetPlacer/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/listener/FleetPlacerQueueListener.java]: Language support for logging.

Comment (Hoymm) [inline FleetPlacer/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/listener/FleetPlacerQueueListener.java]: logInfo varargs for supporting multiple language

File: FleetPlacer/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/event/FleetPlacedEventListenerTest.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    @BeforeMethod
+    public void setUp() {
+        player = new Player("", "");
+        queueService = mock(QueueService.class);
+        fleetPlacedEventListener = new FleetPlacedEventListener(queueService);
+    }
+
+    public void triggerEventOnce_sendObjectToQueue_7() {
+        FleetPlacement fleetPlacement = new FleetPlacement(player, new Fleet());
+        FleetPlacedEvent fleetPlacedEvent = new FleetPlacedEvent(this, fleetPlacement);
```
Comment (Hoymm) [inline FleetPlacer/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/event/FleetPlacedEventListenerTest.java]: Duplicated code :
"FleetPlacement fleetPlacement = new FleetPlacement(player, new Fleet());
 FleetPlacedEvent fleetPlacedEvent = new FleetPlacedEvent(this, fleetPlacement);"

in both methods

File: FleetPlacer/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/event/FleetPlacedEvent.java (hunk 1 above)
Comment (xKoem) [reply FleetPlacer/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/event/FleetPlacedEvent.java]: done

File: FleetPlacer/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/event/FleetPlacedEventListener.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.stereotype.Component;
+import pl.jakubowskiprzemyslaw.tajgertim.models.board.FleetPlacement;
+import pl.jakubowskiprzemyslaw.tajgertim.models.confirmation.FleetPlacementConfirmation;
+import pl.jakubowskiprzemyslaw.tajgertim.queues.Queues;
+import pl.jakubowskiprzemyslaw.tajgertim.services.QueueService;
+
+@Component
+public class FleetPlacedEventListener implements ApplicationListener<FleetPlacedEvent> {
+
+    private int amountOfFleetsConfigured = 0;
```
Comment (totine) [inline FleetPlacer/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/fleetplacer/event/FleetPlacedEventListener.java]: to consideration - without initialization, initialization in constructor?

**Pull request merged**